src/services/knowledge_filter_service.py

```
@@ -35,6 +86,36 @@ async def create_knowledge_filter(
+                refresh="wait_for",
+            )
+            if result.get("result") in ["created", "updated"]:
+                try:
+                    await opensearch_client.indices.refresh(
+                        index=KNOWLEDGE_FILTERS_INDEX_NAME
+                    )
+                except Exception:
+                    pass
+                return {"success": True, "id": filter_doc["id"], "filter": filter_doc}
+            return {"success": False, "error": "Failed to upsert knowledge filter"}
         except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)